vortex-array/src/array/chunked/compute/slice.rs

```diff
@@ -0,0 +1,37 @@
+use vortex_error::VortexResult;
+
+use crate::array::chunked::ChunkedArray;
+use crate::compute::slice::{slice, SliceFn};
+use crate::{ArrayDType, IntoArray, OwnedArray};
+
+impl SliceFn for ChunkedArray<'_> {
+    fn slice(&self, start: usize, stop: usize) -> VortexResult<OwnedArray> {
+        let (offset_chunk, offset_in_first_chunk) = self.find_chunk_idx(start);
+        let (length_chunk, length_in_last_chunk) = self.find_chunk_idx(stop);
+
+        if length_chunk == offset_chunk {
+            if let Some(chunk) = self.chunk(offset_chunk) {
+                return ChunkedArray::try_new(
+                    vec![slice(&chunk, offset_in_first_chunk, length_in_last_chunk)?],
+                    self.dtype().clone(),
+                )
+                .map(|a| a.into_array());
+            }
+        }
+
+        let mut chunks = (offset_chunk..length_chunk + 1)
+            .map(|i| self.chunk(i).unwrap())
```

Comment:
prefer to use expect with a message explaining why this is safe. `.expect("find_chunk_idx returned an incorrect idx")`

Reply:
ok, before it was infallible because I could get the list of all chunks while now it's not explicitly possible.